tests/integration/user_test.go

```diff
@@ -292,8 +292,25 @@ func TestUserLocationMapLink(t *testing.T) {
 	})
 	session.MakeRequest(t, req, http.StatusSeeOther)
 
-	req = NewRequest(t, "GET", "/user2/")
-	resp := session.MakeRequest(t, req, http.StatusOK)
-	htmlDoc := NewHTMLParser(t, resp.Body)
-	htmlDoc.AssertElement(t, `a[href="https://example/foo/A%2Fb"]`, true)
+	// Testing every single tab, despite the fact that they all use
+	// `shared/user/profile_big_avatar.tmpl` (as of the time of this writing)
+	// may be a bit superfluous, but but it is meant to act as a safeguard
+	// for possible future regressions.
+
+	tabsToTest := []string{
+		"/user2",
+		"/user2/?tab=projects",
+		"/user2/?tab=packages",
```

Comment:
I do not think it is right. There is no `tab=packages`.

The test code doesn't really test ....

And I do not think it needs such test since the code in in shared function already.

Reply:
Seems like I messed up something big time as I was trying to get the test to work somehow. Apart from that, the intention was to create a test that checks whether this disabled-by-default feature will still work after a refactor - I know that the code is in a shared function and that the code should work 100% of the time when checking out a specific revision, but, considering that tests are used for development purposes only - are you sure that it doesn't make sense to check for regressions (as stated in the comment)?

Reply:
Have you ever seen `?tab=packages` anywhere? It's not a Gitea URL.

Reply:
Yes, that's indeed a mistake. I'll just bring up that I have had a recent habit of submitting partially faulty PRs over the past few months because of limited time influencing my changes in ways I hadn't anticipated, so I wanted to make a test along with it so as to make sure that I (or someone else, as this feature is not enabled by default) cannot break it.

Unfortunately, it seems my attempt at trying to achieve perfection got in the way of good and that this test is indeed fundamentally incorrect - and you're right about the change being potentially too redundant with the tabs. I'll gladly remove the test so as to not stall this fix from getting merged any further, but do you think that it's worth rethinking *any* form of testing here or just dropping it completely?

Reply:
I (personally) consider core (fundamental)  / fragile-code tests are necessary, while I am neutral for other tests at the moment: good to have if they are correct but not a must. Because I do not see a feasible approach to let every contributor write tests.

For this case, it seems not easy to make it "correct". For example, even if it tests a valid page "/?tab=stars" for current code, the test could still become a no-op in the future: if the "stars" page is refactored to "/-/stars", then "/?tab=stars" shows the default "repositories" page, then the test for "/?tab=stars" still passes but it doesn't really test the new "stars" page.

The root problem is that neither the backend code nor the test code is strict enough. If either of them is strict, then the test makes sense (strict backend: do not accept unknown "tab"; strict test: make sure the tested page is a "stars" page).